rootfs/etc/templates/haproxy/haproxy.tmpl

```diff
@@ -737,11 +737,14 @@ backend {{ $backend.ID }}
 {{- /*------------------------------------*/}}
 {{- range $i, $cors := $corsCfg.Items }}
 {{- if and $cors.Enabled $cors.AllowOrigin }}
-{{- $dynamicOrigin := gt (len $cors.AllowOrigin) 1 }}
+{{- $dynamicOrigin := or $cors.AllowOriginRegex (gt (len $cors.AllowOrigin) 1) }}
 {{- if $dynamicOrigin }}
 {{- range $a1 := short 5 $cors.AllowOrigin }}
     acl cors_allow_origin{{ $i }} var(txn.hdr_origin{{ $i }}) -m str{{ range $a := $a1 }} {{ $a }}{{ end }}
 {{- end }}
+{{- range $a1 := short 5 $cors.AllowOriginRegex }}
+    acl cors_allow_origin{{ $i }} var(txn.hdr_origin{{ $i }}) -m reg{{ range $a := $a1 }} {{ $a |haquote }}{{ end }}
+{{- end }}
```

Comment:
Templating is tricky and breaks very easily, I'd love to see a unit test for this change.

Have a look at this snippet:

https://github.com/jcmoraisjr/haproxy-ingress/blob/5677e338eebef8de5bdde6dcbb3d51c55eb1acb3/pkg/haproxy/instance_test.go#L239-L267

There are some other similar snippets above it with cors tests. Take one of them as an example to create a new one just to cover the new option, even better if you provide at least 6 regex as input.

Reply:
Thanks for pointing me to the right place, added a few tests there.